### DISCUSSION TITLE
DOOM workspaces not compatible

### DISCUSSION THREAD

**Discussion by domodoiddd** [alphapapa/activities.el#91] [General]
I use standard configuration for activities.el, so I expected to it to work in tabs only, but it somehow breaks the workspaces which are provided in doom emacs. So when I create 2 or more workspaces , on each workspace I see the same tabs, when I open one tab in one workspace this tab opens on other workspaces too, even tabs without declared activity are shared among these workspaces. Also in activities-switch-buffers I can see buffers, which were opened not only in one tab activity, but all buffers which were opened in frame. Can I have workspace functionality with activities.el? How to solve issues described above?

**Comment by alphapapa** [comment]
I don't use DOOM and can't offer support for it.  DOOM does things its own way; things are not designed to integrate with DOOM, but vice versa.  You'll have to consult with DOOM developers and users, or disable its workspace functionality and use Activities instead.

**Reply by domodoiddd** [reply]
But in theory, should activities.el work with packages which provide workspaces? If yes, may be I will switch to vanilla emacs then, or I can just turn of doom emacs configurations for workspaces and add my own. May be activities.el can provide one more layer of abstraction? May be it could save/restore at the same time separate workspaces and tabs?

**Reply by alphapapa** [reply]
There is no definition of "workspace" in Emacs's terms.  It's a vague concept that's used in various ways by various tools.  Activities provides its own tools based on the concepts described in its documentation.  It's up to users (and possibly other tools, like DOOM) to use it if desired.

If you have concrete suggestions or ideas, please mention them, but they must be specific to be useful.

**Comment by domodoiddd** [comment]
@alphapapa 
Okay, so here is my idea how we can implement a thing which I call "workspaces", but as you said there is no such thing in emacs terms and you are right. So a workspace is just a group of tabs, this group can consist of one tab too. Activities.el today have two main features: activities per frame, activities per tab. The activities per tab in my opinion can be easily used for this future "workspace" functionality.
So we create new functions like: activities-group or activities-groups, activities-groups-list, activities-group-create, activities-group-delete or activities-group-remove, activities-groups-switch, activities-group-edit(so we can add more tab activities or rename this group, or we can have separate functions for these tasks) etc.
The basic idea of all of these functions is that we use the already existing functionality of activities.el per tabs, so we can just combine/join a list of activities( per tab, hope you understand this) into a group. 
So for example I am working on my emacs configuration, I created one tab activity for doom emacs, second tab activity for spacemacs, one more tab for my own emacs configuration and may be I can even have some other emacs configurations for each tab, so I have 3 or more tabs really for one simple thing, I just want to extend and make better my own emacs configuration, but it is not comfortable to load each tab activity every time, I want to group these tab activites into "one thing" which we can call a "workspace" or a group, so with new functions I could just use function activities-group-create and I could call this group "creating my emacs configuration" or may be just "emacs conf" and just add tabs(activities) to this group.
What do you think about it? Is is possible? I hope, that it will be really easy to implement

So when I switch between groups I will see different tab acitivites in my tabbar, may be we can even write a function for a doom modeline which will show the active activities-group, or we can just provide a simple configuration for modeline in README.md or may be it is even possible to have a name of active group in tab bar itself.

**Reply by alphapapa** [reply]
Please see #84 for discussion about implementing groups of activities, or sub-activities, etc.

And please understand that these features are not so easy to implement--or, if the implementation itself is not difficult, the designing is.  I used Emacs for years thinking about these ideas before I came up with this library.  I don't want to just throw something together on top of what exists.  Slow and steady progress is the goal, because this library should be usable for many years to come.

In the meantime, you're free to implement something like that in your own code and use it; as the discussion on #84 mentions, just looping through a list of activities and opening them isn't difficult.  The hard part is designing a way to do these things simply, in a way users can easily understand, without having a large number of commands to remember or page through, and without interfering with other functionality.
